### PR TITLE
feat: add ability to undo moved document

### LINKED ISSUE #1823: Ability to prevent/undo accidental drag and drop in sidebar
**To Reproduce**
1. Drag Document Accidentally
2. Drop in Misplace
3. See error

**Expected behavior**
Enable or Disable document Drag & Drop in Order menu

**Screenshots**
image

**Outline (please complete the following information):**
- Install: SelfHosted
- Version: 0.52.0

### PR DESCRIPTION
**Pull request by iamsaumya**
closes #1823

### CODE REVIEW

**Comment by iamsaumya** [comment]
For now, it looks much easier than I imagined.

**Comment by iamsaumya** [comment]
@tommoor, the Action didn't look very polished, so I changed it up a bit. Please take a look.

New

https://user-images.githubusercontent.com/40148448/153698030-966f85d2-8e7f-4a8c-81a1-0a1d2c590887.mov


https://user-images.githubusercontent.com/40148448/153698033-617a5d1a-3414-423f-bd7a-e2d065b63236.mov



Old
Screenshot 2022-02-12 at 10 30 15 AM
Screenshot 2022-02-12 at 10 30 06 AM

File: app/components/Sidebar/components/DocumentLink.tsx [hunk 1]
```diff
@@ -182,10 +181,13 @@ function DocumentLink(
...
         return;
       }
+
       if (!collection) {
         return;
       }
-      documents.move(item.id, collection.id, node.id);
+
+      const document = documents.get(item.id);
+      document?.moveWithUndo(collection.id, node.id);
```
Comment (tommoor) [inline app/components/Sidebar/components/DocumentLink.tsx:190]: `moveWithUndo` doesn't exist anymore right?

File: app/components/Sidebar/components/DocumentLink.tsx [hunk 2]
```diff
@@ -248,12 +250,14 @@ function DocumentLink(
...
-      if (expanded) {
-        documents.move(item.id, collection.id, node.id, 0);
-        return;
-      }
-
-      documents.move(item.id, collection.id, parentId, index + 1);
+      const parentDocumentId = expanded ? node.id : parentId;
+      const droppedDocumentIndex = expanded ? 0 : index + 1;
+      const document = documents.get(item.id);
+      document?.moveWithUndo(
```
Comment (tommoor) [inline app/components/Sidebar/components/DocumentLink.tsx:256]: I feel like for reordering the toast and undo is probably overkill? Definitely feels more important when re-parenting

**Comment by iamsaumya** [comment]
Sry, I haven’t worked on this after our chat. I will complete it and request a review.

**Comment by tommoor** [comment]
Spoke with @iamsaumya – we've decided to cancel this feature for now

**Pull request closed without merging**